backend-utils/src/lib.rs

```diff
@@ -241,8 +259,21 @@ fn split_by_namespace<F: FieldElement>(
                     },
                 }
             }
-            StatementIdentifier::ProverFunction(_)
-            | StatementIdentifier::TraitImplementation(_) => None,
+            StatementIdentifier::ProverFunction(i) => {
+                let prover_function = pil.prover_functions.get(*i).unwrap();
+                let namespaces = referenced_namespaces_parsed_expression(prover_function)
+                    .into_iter()
+                    .filter(|namespace| !namespace.starts_with("std::"))
+                    .collect::<BTreeSet<_>>();
+                match namespaces.len() {
+                    1 => Some((namespaces.into_iter().next().unwrap().clone(), statement)),
+                    0 => panic!("Prover function references no namespace: {prover_function}"),
```

Comment:
Should this really be a panic?

Reply:
I mean it could call another function right away, but yeah, probably it would reference at least one column.